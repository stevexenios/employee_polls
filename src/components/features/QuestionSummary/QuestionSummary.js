import { Button, Divider, Modal } from 'antd';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import QuestionTable from '../QuestionTable/QuestionTable';

const QuestionSummary = ({ restore, question, styling }) => {
  const navigate = useNavigate();

  const [displayModal, setDisplayModal] = useState(false);
  const [confirmedAnswer, setConfirmedAnswer] = useState({});
  const [confirmLoading, setConfirmLoading] = useState(false);

  const handleOk = () => {
    if (confirmedAnswer !== {}) {
      setConfirmLoading(true);
      restore(confirmedAnswer);
      setTimeout(() => {
        setDisplayModal(false);
        setConfirmLoading(false);
      }, 500);
    } else
      console.log(
        'ConfirmedAnswer is still an empty object. Nothing to submit!'
      );
  };

  const handleCancel = () => {
    setDisplayModal(false);
  };

  const getTimeElement = (timestamp) => {
    const dateObject = new Date(timestamp);
    const dateString = dateObject.toLocaleDateString('en-US');
    const timeString = dateObject.toLocaleTimeString('en-US');

    return (
      <span className="date-display">
        <i>{timeString}</i> | <b>{dateString}</b>
      </span>
    );
  };

  const goToQuestion = () => {
    navigate(`/questions/${question?.id}`);
  };

  return (
    <div className={'question-container-' + styling}>
      <Divider>
        <h4>{question?.author}</h4>
      </Divider>
      {getTimeElement(question?.timestamp)}
      <Button
        type="primary"
        size="small"
        className="button-display"
        onClick={() => setDisplayModal(true)}
      >
        Display summary
      </Button>

      <Button
        type="primary"
        size="small"
        className="button-display"
        onClick={() => goToQuestion()}
      >
        Go to question
      </Button>

      <Modal
        title="Would You Rather...?"
        open={displayModal}
        okText="Confirm"
        cancelText="Close"
        width="auto"
        onCancel={handleCancel}
        onOk={handleOk}
        confirmLoading={confirmLoading}
      >
        <QuestionTable
          setConfirmedAnswer={setConfirmedAnswer}
          question={question}
          styling={styling}
        />
      </Modal>
    </div>
  );
};

export default QuestionSummary;
