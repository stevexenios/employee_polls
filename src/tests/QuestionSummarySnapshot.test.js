import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from "@testing-library/react";
import QuestionSummary from '../components/features/QuestionSummary/QuestionSummary';

describe('QuestionSummary', () => {
    it('will match snapshot', () => {
        const view = render(
            <BrowserRouter>
                <QuestionSummary />
            </BrowserRouter>
        );
        expect(view).toMatchSnapshot();
    })
});