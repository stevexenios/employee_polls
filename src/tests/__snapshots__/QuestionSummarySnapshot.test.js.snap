// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionSummary will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="question-container-undefined"
      >
        <div
          class="ant-divider css-dev-only-do-not-override-acm2ia ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center"
          role="separator"
        >
          <span
            class="ant-divider-inner-text"
          >
            <h4 />
          </span>
        </div>
        <span
          class="date-display"
        >
          <i>
            Invalid Date
          </i>
           | 
          <b>
            Invalid Date
          </b>
        </span>
        <button
          class="ant-btn css-dev-only-do-not-override-acm2ia ant-btn-primary ant-btn-sm button-display"
          type="button"
        >
          <span>
            Display poll
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-acm2ia ant-btn-primary ant-btn-sm button-display"
          type="button"
        >
          <span>
            Question Details
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="question-container-undefined"
    >
      <div
        class="ant-divider css-dev-only-do-not-override-acm2ia ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center"
        role="separator"
      >
        <span
          class="ant-divider-inner-text"
        >
          <h4 />
        </span>
      </div>
      <span
        class="date-display"
      >
        <i>
          Invalid Date
        </i>
         | 
        <b>
          Invalid Date
        </b>
      </span>
      <button
        class="ant-btn css-dev-only-do-not-override-acm2ia ant-btn-primary ant-btn-sm button-display"
        type="button"
      >
        <span>
          Display poll
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-acm2ia ant-btn-primary ant-btn-sm button-display"
        type="button"
      >
        <span>
          Question Details
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
